### PR TITLE
Tags should refer to full paths, not just base paths.

### PR DESCRIPTION
**Pull request by adamchalmers**
See "Problem 2" in https://github.com/KittyCAD/modeling-app/issues/4297

This is a pure refactor, it should not change any behaviour at all.
It adds more information into the tag system, but nothing reads that
extra information yet. It will be used to address problem 3 of the above
issue.

### CODE REVIEW

File: src/wasm-lib/kcl/src/executor.rs [hunk 1]
```diff
@@ -1695,6 +1721,21 @@ impl Path {
...
     }
 
+    /// Where does this path segment start?
+    pub fn get_from(&self) -> &[f64; 2] {
+        &self.get_base().from
+    }
+    /// Where does this path segment end?
+    pub fn get_to(&self) -> &[f64; 2] {
+        &self.get_base().to
+    }
```
Comment (lf94) [inline src/wasm-lib/kcl/src/executor.rs:1731]: Why the unnecessary indirection?

**Review by lf94 — APPROVED** [review]
LGTM. I kinda wish we errored when `lengh` is called on a non-linear line but I'm hoping the current method is very temporary. If it's not very temporary (< 1 month) we should do it

**Review by jtran — APPROVED** [review]
Nice.

File: src/wasm-lib/kcl/src/executor.rs (hunk 1 above)
Comment (adamchalmers) [reply src/wasm-lib/kcl/src/executor.rs:1731]: In the future we're not going to know the start/end of lines so easily (we'll need to query wasm or the API) so this change is preparation for that.

**Comment by adamchalmers** [comment]
> LGTM. I kinda wish we errored when `length` is called on a non-linear line but I'm hoping the current method is very temporary. If it's not very temporary (< 1 month) we should do it

will be done today

File: src/wasm-lib/kcl/src/executor.rs (hunk 1 above)
Comment (lf94) [reply src/wasm-lib/kcl/src/executor.rs:1731]: get_base() though is already abstracting

**Pull request merged**